ansible/rke2/airgap/playbooks/deploy/rke2-tarball-playbook.yml

```diff
@@ -0,0 +1,214 @@
+     retries: 3
+     delay: 5
+
+   - name: Verify kubectl installation
+     command: kubectl version --client
+     register: kubectl_version
+     changed_when: false
+
+   - name: Display kubectl version
+     debug:
+       msg: "kubectl installed: {{ kubectl_version.stdout }}"
+
+   - name: Create .kube directory for root
+     file:
+       path: /root/.kube
+       state: directory
+       mode: '0700'
+       owner: root
+       group: root
+
+   - name: Create .kube directory for ansible user
+     file:
+       path: "/home/{{ ansible_user }}/.kube"
+       state: directory
+       mode: '0700'
+       owner: "{{ ansible_user }}"
+       group: "{{ ansible_user }}"
+     when: ansible_user is defined and ansible_user != 'root'
+
+   - name: Fetch KUBECONFIG from first airgap node
+     fetch:
+       src: /etc/rancher/rke2/rke2.yaml
+       dest: /tmp/rke2-{{ inventory_hostname }}.yaml
+       flat: yes
+     delegate_to: "{{ groups['airgap_nodes'][0] }}"
+     become: true
+
+   - name: Read kubeconfig content
+     slurp:
+       src: "/tmp/rke2-{{ inventory_hostname }}.yaml"
+     register: kubeconfig_content
+     delegate_to: localhost
+     become: false
+
+   - name: Update server URL in kubeconfig content
+     set_fact:
+       updated_kubeconfig: "{{ (kubeconfig_content.content | b64decode) | regex_replace('https://127\\.0\\.0\\.1:6443', 'https://' + hostvars[groups['airgap_nodes'][0]]['ansible_host'] + ':6443') }}"
+
+   - name: Write kubeconfig to bastion root
+     copy:
+       content: "{{ updated_kubeconfig }}"
+       dest: /root/.kube/config
+       mode: '0600'
+       owner: root
+       group: root
+
+   - name: Write kubeconfig to ansible user
+     copy:
+       content: "{{ updated_kubeconfig }}"
+       dest: "/home/{{ ansible_user }}/.kube/config"
+       mode: '0600'
+       owner: "{{ ansible_user }}"
+       group: "{{ ansible_user }}"
+     when: ansible_user is defined and ansible_user != 'root'
+
+   - name: Clean up temporary kubeconfig
+     file:
+       path: "/tmp/rke2-{{ inventory_hostname }}.yaml"
+       state: absent
+     delegate_to: localhost
+     become: false
+
+   - name: Test kubectl connectivity
+     shell: kubectl get nodes -o wide
+     environment:
+       KUBECONFIG: /root/.kube/config
+     register: kubectl_test
+     failed_when: false
+     changed_when: false
+
+   - name: Display kubectl test results
+     debug:
+       msg: |
+         kubectl connectivity test:
+         {% if kubectl_test.rc == 0 %}
+         SUCCESS - Cluster accessible from bastion:
+         {{ kubectl_test.stdout }}
+         {% else %}
+         FAILED - Error connecting to cluster:
+         {{ kubectl_test.stderr }}
+         {% endif %}
+
+   - name: Display final instructions
+     debug:
+       msg: |
+         🎉 RKE2 Cluster Setup Complete!
+         
+         ✅ RKE2 cluster deployed on airgap nodes
+         ✅ kubectl installed on bastion node
+         ✅ KUBECONFIG copied and configured
```

Comment:
How do the emojis look in a shell window? It looks good here in review, but I'm curious how it will look from command prompt if I run there, and how it will look in Jenkins' HTML console logs?

Reply:
That's a good question. In my terminal it looks fine. I think any browser will render the emoji's.  But if they are too snazzy I can remove them.